packages/sanity/src/core/releases/tool/components/releaseCTAButtons/ReleaseRevertButton/useDocumentRevertStates.ts

```diff
@@ -0,0 +1,135 @@
+import {type SanityDocument} from '@sanity/types'
+import {useCallback, useEffect, useMemo, useRef} from 'react'
+import {useObservable} from 'react-rx'
+import {catchError, filter, forkJoin, from, map, type Observable, of, switchMap} from 'rxjs'
+
+import {useClient} from '../../../../../hooks/useClient'
+import {getTransactionsLogs} from '../../../../../store/translog/getTransactionLogs'
+import {API_VERSION} from '../../../../../tasks/constants'
+import {type DocumentInRelease} from '../../../detail/useBundleDocuments'
+
+export type RevertDocument = SanityDocument & {
+  _system?: {
+    delete: boolean
+  }
+}
+
+type RevertDocuments = RevertDocument[]
+
+type DocumentRevertStates = RevertDocuments | null | undefined
```

Comment:
Does `null` have a special meaning? If so, would be great to have a short comment in code here explaining what it means, if not, I tend to prefer sticking with just `undefined`